cmd/gapid/main.go

```diff
@@ -149,53 +157,62 @@ func newConfig() *config {
 	return c
 }
 
+func (c *config) logIfVerbose(args ...interface{}) {
+	if c.verbose {
+		fmt.Println(args...)
+	}
+}
+
 func (c *config) locateCWD() error {
 	cwd, err := os.Executable()
 	if err != nil {
 		return err
 	}
 	cwd, err = filepath.EvalSymlinks(cwd)
-	c.cwd = filepath.Dir(cwd)
+	if err == nil {
+		c.cwd = filepath.Dir(cwd)
+		c.logIfVerbose("CWD:", c.cwd)
+	}
 	return err
 }
 
 func (c *config) locateVM() error {
 	if c.vm != "" {
-		if checkVM(c.vm, false) {
+		if c.checkVM(c.vm, false) {
 			return nil
 		}
 
-		if runtime.GOOS == "windows" && checkVM(c.vm+".exe", false) {
+		if runtime.GOOS == "windows" && c.checkVM(c.vm+".exe", false) {
 			c.vm += ".exe"
 			return nil
 		}
 
-		if java := c.javaInHome(c.vm); checkVM(java, false) {
+		if java := c.javaInHome(c.vm); c.checkVM(java, false) {
 			c.vm = java
 			return nil
 		}
-		return fmt.Errorf("JVM '%s' not found", c.vm)
+		return fmt.Errorf("JVM '%s' not found/usable", c.vm)
 	}
 
-	if java := c.javaInHome(filepath.Join(c.cwd, "jre")); checkVM(java, true) {
+	if java := c.javaInHome(filepath.Join(c.cwd, "jre")); c.checkVM(java, true) {
 		c.vm = java
 		return nil
 	}
 
 	if home := os.Getenv("JAVA_HOME"); home != "" {
-		if java := c.javaInHome(home); checkVM(java, true) {
+		if java := c.javaInHome(home); c.checkVM(java, true) {
 			c.vm = java
 			return nil
 		}
 	}
 
-	if java, err := exec.LookPath(c.javaExecutable()); err == nil && checkVM(java, true) {
+	if java, err := exec.LookPath(c.javaExecutable()); err == nil && c.checkVM(java, true) {
 		c.vm = java
 		return nil
 	}
 
 	if runtime.GOOS == "linux" {
-		if java := "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java"; checkVM(java, true) {
+		if java := "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java"; c.checkVM(java, true) {
 			c.vm = java
 			return nil
 		}
```

Comment:
Here, we know we're on Linux and no suitable JVM has been found, this is maybe the best place to help the user and unconditionally log a help message like:
```
} else {
    log.I(ctx, "No suitable JVM found, please install openjdk-8-jre")
}
```

Reply:
Already doing that above. Again, the openjdk-8-jre should already be installed since the .deb package depends on it.